### ISSUE TITLE
How to click on a nested element

### ISSUE THREAD

**Issue by janmeier** [bruderstein/unexpected-react#23]
Hello, it's me again :)

```js
class Todo extends Component {
  render() {
    return (
      <div onClick={() => console.log('div clicked')}>
        <span>{this.props.name}</span>
        <button onClick={() => console.log('button clicked')}>Finish it</button>
      </div>
    );
  }
}

class TodoList extends Component {
  render() {
    let list = [
      { id: 42, name: 'foo' },
      { id: 43, name: 'bar' }
    ];

    return (
      <div>
        {list.map(item => {
          return <Todo key={item.id} {...item} />;
        })}
      </div>
    );
  }
}
expect(
  TestUtils.renderIntoDocument(<TodoList />),
  'with event click', 'on', <Todo id={42} ><button/></Todo>
);
```

Basic todo list example, but instead of clicking the item, you have to click a button. I want to click on the button for a specific item, therefore I wrap `button` in a `Todo` with id. The problem is that this clicks the parent tag of the `Todo`, the div, and not the button. I've also tried different variations of `'queried for', <Todo />`, but that doesn't let me click anything afterwards

**Comment by bruderstein** [comment]
Using the full renderer, this should be possible with `queried for` (note: not with the shallow renderer, yet)

However, I've just tried it, and you're right, it doesn't work.  I see the issue, should have a fix out shortly.

**Comment by bruderstein** [comment]
Fixed in 3.1.2.

You should now be able to do

```js
expect(component, 'queried for', <Todo id={42} />,
    'with event click', 'on', <button />,
    'to have rendered',
    <Todo><span>foo</span></Todo>); 
```

Let me know if that works now :)

**Comment by janmeier** [comment]
Works perfectly, thanks a bunch!
